### ISSUE TITLE
Interpreting output

### ISSUE THREAD

**Issue by caiwjohn** [Sylvie/sambada#21]
Hey! Thanks for all the work you've put in on Sambada. I've managed to get a set of results but am struggling to interpret them. The post-processing section in the documentation seems to be missing. I see log likelihood, G scores, and Wald scores but can't figure out how to assess the significance of these statistics without understanding their distribution.

What is the log likelihood of? Is it a likelihood ratio test? If so, which model is numerator and which is denominator? I'm also not sure how to interpret the significance of the Wald scores. I don't believe I can look at the distribution of Wald scores in the result output because it has already been filtered for significance so we won't have representative sampling.

Any help would be greatly appreciated!

**Comment by Sylvie** [comment]
Hello @caiwjohn,
Thanks a lot for your message and your interest in Sambada!

I must admit that the theoretical background of Sambada is a bit scattered around, and sometimes hidden in on-subscription journals.
Sorry for the inconvenience!

Regarding your questions, yes, the G score computed by Sambada is the score of a likelihood ratio test. The test compares the model with N variables with the "best" model with N-1 variables. If you compute univariate models, each models under analysis is compared to the corresponding null model (with no environmental variables).
(For a discussion of the multivariate analysis, please refer to the section "Materials and methods" of:
https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.12629
and to the first page of the supplementary information.)
Both likelihood ratio and Wald tests use a Chi-squared statistic to assess the significance of the models.
The number of degrees of liberty is the number of variables added in the model to assess compared to the reference model.
If you wish to know how to compute the p-values corresponding to G and Wald scores provided by Sambada, please have a look at
https://github.com/Sylvie/sambada/blob/master/scripts/bonferroniModelSelection.R
The script shows an example on how to select the significant models out of Sambada's results.
The computation of p-values is on line 41.

Regarding the detailed formulae, there is a summary in my thesis, but it is in French ;-)
https://infoscience.epfl.ch/record/196921
See pages 76-77 for the G and Wald scores.

When running Sambada, you can choose to store the results for all models, or only for those which are significant (with the param SAVETYPE). As you mentioned, if you choose the second option, the distributions of G and Wald scores will be cut.
One thing to know is that Sambada applies a Bonferroni correction during model selection to take into account the fact that multiple models are considered. The procedure is described in the section "Univariate analysis" of the article, and an example is provided in the script.
If you prefer to assess the significance of the models by yourself and/or have a look at the distribution of scores, let Sambada store all models, and apply the selection procedure of your choice.

Does this answer your question?

Kind regards,
Sylvie

PS: Thanks a lot for your feedback on the manual, I will try to fill the gap in a future release.

**Comment by caiwjohn** [comment]
Thanks so much Sylvie! Sorry for the delayed response, I was able to get some good results using the software and really appreciate your responsiveness and description. I'm citing you guys in my manuscript now.

As far as the manual goes, there were a couple issues that were the biggest obstacles to me getting the software running. I believe these all apply to samBada but I was also interfacing with the R.samBada package so its possible the CSV issue was on their end.

1. Throughout the manual and software you claim .csv files are the mandatory input format. CSV stands for comma separated values but it seems that your software actually requires white-space separated values (i.e. tab or space delimited files). I see how this might happen as csv has become shorthand for any file structured with rows and columns but as someone more on the comp sci side of things I interpreted csv literally. Obviously I couldn't get the functions to run and it took me a while to figure out that I actually needed to whitespace delimit them.

2. If you maybe just add a sentence to the post-processing section saying that anyone looking to compute p.values from the output can use the bonferroniModelSelection.R script it might help others in the future. That script is great and I was able to use it successfully once you pointed it out.

Thanks again for your help and contribution to the community.
